### ISSUE TITLE
Add support for timezones to the REST API data layer.

### ISSUE THREAD

**Issue by peterfabian** [woocommerce/wc-admin#489]
Currently, the API assumes all times are in UTC, which is not the case, the API should stick to local timezone set in WP settings, and create time buckets accordingly.

**Comment by peterfabian** [comment]
I added my WIP to https://github.com/woocommerce/wc-admin/pull/492. If this needs to be finished before the Woo conference in Oct, it probably should be taken over by @claudiosanches.

**Comment by timmyc** [comment]
@peterfabian do you consider this a blocker for merging into Core?

**Comment by timmyc** [comment]
Wanted to provide an illustration of this bug in action ( at least I assume it is the same bug ) - and some steps to reproduce:

- Place an order today
- View the orders report for Year to Date

![image](https://user-images.githubusercontent.com/22080/52500012-bd700c00-2b91-11e9-8f1b-1ed73d01dd3d.png)

Note the **Orders Count** shows 11 orders, while in the tabular data, we can see there are 12 orders:

![image](https://user-images.githubusercontent.com/22080/52500098-fad49980-2b91-11e9-9f0d-2cddf53c21d1.png)

And vs the legacy report, which shows 12 orders:

![image](https://user-images.githubusercontent.com/22080/52500119-0a53e280-2b92-11e9-97c4-86392bfa6568.png)

If I hack the date range to go till tomorrow ( Feb 9 for this example ) all looks well:

![image](https://user-images.githubusercontent.com/22080/52500167-340d0980-2b92-11e9-90bb-942324af86c5.png)

Note the dollar amounts don't jive between the two reports, I will dig into that and file another issue if I can track it down/reproduce it.

Related `wc_order_stats` entry:

![image](https://user-images.githubusercontent.com/22080/52500373-ad0c6100-2b92-11e9-8ce2-8cc77c820d76.png)

@peterfabian do you think this might be a good issue for @rrennick to take a look at and/or adopt your PR ( #492 )?

**Comment by rrennick** [comment]
This would be a decent one for me to start on. I did a few PRs on the legacy reports so I'm already familiar with some of those.

**Comment by timmyc** [comment]
Nice @rrennick! Also with the start in the PR above, it has some prior art to work from. That PR is pretty out-of-date against `master` so might be easier to make a fresh branch and start over, but totally up to you.

**Comment by peterfabian** [comment]
I'm sorry, I already started reviving this one last week in #1515  so maybe @rrennick can start with a PR review plus possible fixing of the bug described by @timmyc if it's still a bug? If not, we can try to find something else perhaps?
